### PR TITLE
Add syndicate dimension pot

### PR DESCRIPTION
**Pull request by Ilya246**
## About the PR

adds a purchasable pocket dimension for syndicate agents

disguised as a plant, altclick or rightclick and open verb menu to toggle portal

is indestructible so you can't delete people or items

costs 30tc

**Media**

https://user-images.githubusercontent.com/57039557/234670495-2df14b24-b351-4df5-b496-e2f3cf5fb281.mp4

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: deltanedas + Ilya246
- add: Syndicate R&D has retrofitted stolen teleporter tech and invented a portable pocket dimension portal.

### CODE REVIEW

**Comment by HoidC** [comment]
holy based but could it get different item reskins to prevent metagaming power?

**Comment by Ilya246** [comment]
the single sprite is partially there to justify the 12TC price, if the metagaming gets real bad will implement sprite randomisation later yes

**Comment by deltanedas** [comment]
its just alt click so its pretty hard to not metagame

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+/// When activated by alt verb, spawns a portal to this dimension or closes it.
+/// </summary>
+[RegisterComponent]
+[Access(typeof(DimensionPotSystem))]
+public sealed class DimensionPotComponent : Component
+{
+    /// <summary>
+    /// The portal on the pot, if it is open right now.
+    /// </summary>
+    [DataField("potPortal")]
```
Comment (deltanedas) [inline Content.Server/Teleportation/Components/DimensionPotComponent.cs:21]: change this to just be [ViewVariables]

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    public EntityUid? PotPortal = null;
+
+    /// <summary>
+    /// The portal in the pocket dimension, usually exists.
+    /// </summary>
+    [DataField("dimensionPortal")]
```
Comment (deltanedas) [inline Content.Server/Teleportation/Components/DimensionPotComponent.cs:27]: ,

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    [DataField("potPortalPrototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string PotPortalPrototype = "PortalRed";
+
+    /// <summary>
+    /// The prototype to spawn for the portal spawned in the pocket dimension.
+    /// </summary>
+    [DataField("dimensionPortalPrototype", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
+    public string DimensionPortalPrototype = "PortalBlue";
+
+    [DataField("openPortalSound")]
```
Comment (deltanedas) [inline Content.Server/Teleportation/Components/DimensionPotComponent.cs:54]: these sounds should be either played when opening/closing portal or removed

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    public SoundSpecifier OpenPortalsSound = new SoundPathSpecifier("/Audio/Machines/high_tech_confirm.ogg")
+    {
+        Params = AudioParams.Default.WithVolume(-2f)
+    };
+
+    [DataField("closePortalSound")]
+    public SoundSpecifier ClearPortalsSound = new SoundPathSpecifier("/Audio/Machines/button.ogg");
```
Comment (deltanedas) [inline Content.Server/Teleportation/Components/DimensionPotComponent.cs:61]: Clear -> Close pro

File: Content.Server/Teleportation/Systems/DimensionPotSystem.cs [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    [Dependency] private readonly MapLoaderSystem _mapLoader = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly IMapManager _mapMan = default!;
+
+    private ISawmill _sawmill = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<DimensionPotComponent, ComponentStartup>(OnStartup);
```
Comment (deltanedas) [inline Content.Server/Teleportation/Systems/DimensionPotSystem.cs:29]: base.Initialize(); at the start

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by Ilya246** [comment]
should've probably mentioned that this is ready for review

**Comment by keronshb** [comment]
This is an interesting concept but I feel this might work better as an item for wizard or something.

@Ilya246 merge master and rerun tests please.

**Comment by Ilya246** [comment]
when wizard is a thing could make a version with a considerably bigger pocket dimension, maybe could dynamically generate the pocket dimension too (using a size parameter) since it's just a square with some components added

**Comment by Ilya246** [comment]
fixed, should pass tests now (checked locally), retested ingame and all works well

**Comment by Mudsson** [comment]
this will be very fun and interesting. I can already see entire armorys dissapearing up the pot. Can nukies purchase it? Hiding bodies will be good too. but my main question is, can it be sold? If cargo could sell it, then will that destroy bodies? WIll it include the value of the bluespace storage dimension?

**Comment by Ilya246** [comment]
honestly yeah i should just blacklist it from being sold

**Comment by Ilya246** [comment]
only fails the map schema validator check and the fail isn't even caused by the PR so should be ready for review once again

**Comment by AJCM-git** [comment]
Yeah this is more of a wizard power, exceptions can be made about magic-like items for traitors if its funny (like hot potato) but that isnt the case here, you can re-implement this as a spellbook which shouldnt be that bad  to port from what i see in the code

**Comment by Ilya246** [comment]
i still think this should be a traitor item since it would introduce way more interesting gameplay if it were added as a traitor item as opposed to being a wizard item, there's a much larger variety of uses for it this way too

**Comment by HoidC** [comment]
> Yeah this is more of a wizard power, exceptions can be made about magic-like items for traitors if its funny (like hot potato) but that isnt the case here, you can re-implement this as a spellbook which shouldnt be that bad to port from what i see in the code

this is a really cool traitor item that could provide a lot of utility in gameplay though

wizard doesn't even exist rn and I don't see why they'd ever buy this anyway if they could

**Comment by deltanedas** [comment]
can always just make this a wizard item once someone implements wizard

**Comment by keronshb** [comment]
> i still think this should be a traitor item since it would introduce way more interesting gameplay if it were added as a traitor item as opposed to being a wizard item, there's a much larger variety of uses for it this way too

My issue with this as a traitor item is that it's an easily meta changing item.

If it's in as it is now it should be a 21-25 TC item so it requires cooperation from other traitors due to how powerful it is.

These are two ways where it could powergame hard:

https://user-images.githubusercontent.com/54602815/235541142-f99e0581-8766-46ea-8851-cc52aa852d06.mp4

Being able to zip back and forth between a portal is completely awful for anyone chasing you. There's no delay so someone could just keep going in and out.

Also any canister opened in there is a free near instant kill from pressure/gas or fire.

If the plant or portal is somehow deleted it will also delete the player and everything in the room. Something like the Recycler could possibly be upgraded to grind larger items like it does in SS13 or whenever the Supermatter is implemented it could dust.

If someone spaces the plant when someone or something is inside, they are effectively taken out of the round.

This is why it would be better for wizard/magic. Even then it'd still have to be adjusted but it makes more logistical sense for a wizard to have this ability. Eventually this could be expanded to something like Ninja (who had a capture room at some point) or Clock Cult (who can summon Reebe on some servers as a pocket dimension).

That being said, there is an item in SS13 called the Bluespace Desynchronizer. It's locked behind syndicate/alien technology and it's a toggle that makes you disappear into a "different dimension" for up to 5 minutes. And by disappear you just literally disappear and can't move. It's pretty broken but no one really abused it to high hell since it was such a niche item. This could also be a replacement for that with some modifications.

**Comment by deltanedas** [comment]
> Ninja (who had a capture room at some point) or Clock Cult (who can summon Reebe on some servers as a pocket dimension).

im not adding the dojo, learning from ss13s mistakes

**Comment by deltanedas** [comment]
> Also any canister opened in there is a free near instant kill from pressure/gas or fire.

this can be fixed in general by making portals also teleport gases, makes no sense really that it doesnt

> If the plant or portal is somehow deleted it will also delete the player and everything in the room. Something like the Recycler could possibly be upgraded to grind larger items like it does in SS13 or whenever the Supermatter is implemented it could dust.

i tried implementing ejecting things when it dies before and it was *really* crashy
didnt think about supermatter tho, good point

**Comment by keronshb** [comment]
> this can be fixed in general by making portals also teleport gases, makes no sense really that it doesnt

That would also in turn make it worse. Having a hidden plasma flood/fire isn't fun.

**Comment by metalgearsloth** [comment]
Speaking from DND portable holes are extremely busted.

**Comment by Ilya246** [comment]
could just make the pocket dimension permanently have default airmix
as for deletion/spacing the pot, you'd need to have the target somehow remain in the pot while you move it to deletion machine of choice, in which case it'd be better to just delete/space the target without using the pot - i doubt any non-syndicate you want to kill would just agree to go into the pot and remain there, and if they do, it'd be an easy kill either way
as for the chasing issue, there'd have to conveniently be a dimension pot on your escape route for that to actually happen, you could also just buy it on the spot but you'd be better off just buying an esword
anyway i could raise the price to 14-15tc

**Comment by Ilya246** [comment]
changes have been made (and tested) and i think that:
1) primary addressed balance issues have been fixed, those being a) using custom gas inside the pot to instakill people b) the pot being too cheap
2) i already said this in my comment above, but other addressed balance issues describe situations where you'd be better off just using another option (esword, just doing it without the pot), especially with the 12->14TC price change and the resulting decrease of the variety of items you can buy with it (need renault or another syndie to get esword, emag, or some other items with it)

**Comment by keronshb** [comment]
> as for deletion/spacing the pot, you'd need to have the target somehow remain in the pot while you move it to deletion machine of choice, in which case it'd be better to just delete/space the target without using the pot - i doubt any non-syndicate you want to kill would just agree to go into the pot and remain there,

You're underestimating how easy this can be.

Stun > cuff > chair > build walls around them

**Comment by Ilya246** [comment]
> Stun > cuff > chair > build walls around them > drag pot to space/supermatter
Or
Stun > cuff > locker > lock/weld
>
or just drag the cuffed target without using the pot, potentially in a locker/crate

**Comment by keronshb** [comment]
> > Stun > cuff > chair > build walls around them > drag pot to space/supermatter
> > Or
> > Stun > cuff > locker > lock/weld
> 
> or just drag the cuffed target without using the pot, potentially in a locker/crate

The entire point is you can put someone into the pot and have them silently deleted.

**Comment by Ilya246** [comment]
> The entire point is you can put someone into the pot and have them silently deleted.
>
or you can gib them

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 6]
```diff
@@ -290,6 +290,16 @@
   categories:
     - UplinkUtility
 
+- type: listing
+  id: UplinkDimensionPot
+  name: uplink-dimension-pot-name
+  description: uplink-dimension-pot-desc
+  productEntity: DimensionPot
+  cost:
+    Telecrystal: 14
```
Comment (keronshb) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:299]: Needs to be 20 or higher
```suggestion
    Telecrystal: 20
```

**Review by keronshb — CHANGES_REQUESTED** [review]

**Comment by deltanedas** [comment]
yeah this is way too much effort and tc to delete someone when you can just gib and juice organs or throw everything into space

**Comment by deltanedas** [comment]
also wrt supermatter... just drag the bozo into the crystal? its no more sus than deleting a potted plant

**Comment by keronshb** [comment]
> > The entire point is you can put someone into the pot and have them silently deleted.
> 
> or you can gib them, you can buy suspicious toolbox to do it fast

You're missing the point of people will use this as a tool to easily round remove people in a dimension where they cannot be recovered if the pot is deleted and it's completely busted for chases.

Citadel had something similar called a bluespace jar. Way more busted and could be freely printed. But security and traitors used it to jail people infinitely in a round. Or alternatively they used it as a pocket army.

**Comment by Ilya246** [comment]
i don't have any experience with anything similar so i'll just go with what you said and set the price to 20 as requested, if we end up being correct on 20 TC being too much the price can be reduced later

**Comment by Ilya246** [comment]
again, i don't think the round removal thing is not a new issue adding the pot would create nor an issue it would make worse, you can just either 1) gib 2) put body into a locker/crate, the pot doesn't help in any way here as using the pot has no advantage over using a crate/locker
it should be kept in mind that people will immediately metagame the pot if they see you dragging it

**Comment by Ilya246** [comment]
was the misunderstanding in that you thought the internal portal is toggleable? because it's not, it stays open at all times so you can't trap people inside and can always leave the dimensio

**Comment by keronshb** [comment]
> 

You're not listening on why it's an issue.

It's not an issue because the portal doesn't close.

it's an issue because it makes it an easier tool to do round removal.

YES you can cuff someone to a chair and build walls around them on any map.. but someone is bound to find them.

With the plant you can do the above, then just turn the portal off and stuff the plant somewhere inconspicuous. Very easy cheesy round removal + baby jail.

YES you can just gib someone/do whatever. But the plant adds an extra step to that of having it be nearly untraceable.

This also causes issues of things like the nuke disk or other important items going missing and having them in a pocket dimension isn't the best thing for gameplay.

**Comment by Ilya246** [comment]
> YES you can cuff someone to a chair and build walls around them on any map.. but someone is bound to find them.
With the plant you can do the above, then just turn the external portal off and stuff the plant somewhere inconspicuous. Very easy cheesy round removal + baby jail.

i'd expect people to metagame the plant much more than expected here, which makes hiding it not be much better than alternatives: if you can hide the plant well enough, you can hide a locker/crate/just the body well enough
while this issue does somewhat apply, i think even a cost as low as 12 TC would justify having it
also, i'd argue with `but someone is bound to find them`, people wouldn't just randomly deconstruct specifically the walls surrounding said chair
> YES you can just gib someone/do whatever. But the plant adds an extra step to that of having it be nearly untraceable.

i don't think this is true, if you can somehow find a way to delete the plant that means you can drag the pot to singulo/whatever deletion machine and toss it inside without being spotted, at which point you might as well just drag the body itself or a crate/locker
and again, i think the cost justifies this too: you're spending a large amount of (12 to 20) TC to unreliably roundremove someone while a minibomb (6 TC) or even just a syndie toolbox (4 TC) can do just as well with comparable or potentially even better (don't have to drag thing all the way to deletion machine) effort/discovery chance
as for the untraceability part, i don't see how gibbing is any more traceable: yes, there will be gibbed remains (and maybe a minibomb hole) in maints, so what? if you consider potentially raising the alert level into account, i, again, think the cost would justify this: and again, you could use a crate/locker equally as easily as the pot for dragging the body into a deletion machine
> This also causes issues of things like the nuke disk or other important items going missing and having them in a pocket dimension isn't the best thing for gameplay.

i'd argue that the pocket dimension has a fair chance to be found and people wouldn't use it for that, and besides: 1) storage implant is already a thing 2) you'd have to actually go retreive the item for the objective to count

OVERALL: i think the issues presented are exaggerated as 1) the pot would be metagamed much more than expected (see: chameleon, voicemask, PDA uplink, etc.) 2) the advantage of using the pot over alternatives is exaggerated

**Comment by retequizzle** [comment]
got brought up in admin chat so just going to drop this off here:

OkCICeX

tl;dr for the picture:
- price point is way too low as is
- there needs to be some sort of limitation on how much can be stuffed in there

ideally i'd still only like to see this limited to wizards once they're fully implemented (just a week away) but i also recognize that they're not in the game yet and can at least see why people would want this added as a generic traitor item. but 12-15 TC is simply too low and i will not budge on that. initially i said 40+ but i'd at least be willing to try 28-30 TC minimum and see how much use it gets.

**Comment by Ilya246** [comment]
could this be merged at a price of 30 TC and potentially have the price reduced in a future PR if it ends up not being used at all or ends up having real ingame usages be way too weak for its price?

**Comment by retequizzle** [comment]
agreed with mirror, would be fine with testing that and seeing where it goes

**Comment by Ilya246** [comment]
as seen in the recent commit, price has been reduced to 30 TC as discussed

File: Resources/Prototypes/Catalog/uplink_catalog.yml [hunk 7]
```diff
@@ -290,6 +290,16 @@
...
+- type: listing
+  id: UplinkDimensionPot
+  name: uplink-dimension-pot-name
+  description: uplink-dimension-pot-desc
+  productEntity: DimensionPot
+  cost:
+    Telecrystal: 30
+  categories:
+  - UplinkUtility
+
```
Comment (keronshb) [inline Resources/Prototypes/Catalog/uplink_catalog.yml:302]: Blacklist this from the nukie uplink

File: Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml [hunk 8]
```diff
@@ -265,3 +265,16 @@
...
     state: plant-29
+
+- type: entity
+  id: DimensionPot
+  parent: PottedPlant26
+  suffix: Dimension
+  components:
+  - type: DimensionPot
+  # don't want the pot going into its own portal do we
+  - type: PortalBlacklist
```
Comment (keronshb) [inline Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml:276]: Add this to construction objects so you can't drag things in like wall girders

File: Resources/Maps/Misc/pocket_dimension.yml [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+  author: Ilya246
+  postmapinit: false
+tilemap:
+  0: Space
+  55: FloorPlanetGrass
+entities:
+- proto: ""
+  entities:
+  - uid: 1
+    components:
```
Comment (keronshb) [inline Resources/Maps/Misc/pocket_dimension.yml:13]: Blacklist construction so people can't build inside the dimensions

**Review by keronshb — CHANGES_REQUESTED** [review]
A few more things to change. Someone else will be reviewing the map generation part.

File: Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml (hunk 8 above)
Comment (deltanedas) [reply Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml:276]: you can still build anything inside so what would this intend to prevent

File: Resources/Maps/Misc/pocket_dimension.yml (hunk 9 above)
Comment (deltanedas) [reply Resources/Maps/Misc/pocket_dimension.yml:13]: is there just a component added to the BecomesStation entity?

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    /// The portal in the pocket dimension, usually exists.
+    /// </summary>
+    [ViewVariables]
+    public EntityUid? DimensionPortal = null;
+
+    /// <summary>
+    /// Map of the pocket dimension, usually exists.
+    /// </summary>
+    [ViewVariables]
+    public MapId PocketDimensionMap = MapId.Nullspace;
```
Comment (metalgearsloth) [inline Content.Server/Teleportation/Components/DimensionPotComponent.cs:34]: Needs datafields and pocketdimensionmap should probably be an entityuid.

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+    /// <summary>
+    /// Path to the pocket dimension's map file
+    /// </summary>
+    [DataField("pocketDimensionPath")]
+    public string PocketDimensionPath = "/Maps/Misc/pocket_dimension.yml";
```
Comment (metalgearsloth) [inline Content.Server/Teleportation/Components/DimensionPotComponent.cs:40]: ResPath

File: Content.Server/Teleportation/Systems/DimensionPotSystem.cs [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+		base.Initialize();
+        SubscribeLocalEvent<DimensionPotComponent, ComponentRemove>(OnRemoved);
+        SubscribeLocalEvent<DimensionPotComponent, GetVerbsEvent<AlternativeVerb>>(AddTogglePortalVerb);
+
+        _sawmill = Logger.GetSawmill("dimension_pot");
+    }
+
+    private void OnRemoved(EntityUid uid, DimensionPotComponent comp, ComponentRemove args)
+    {
+        _sawmill.Warning($"Somehow destroyed pocket dimension {comp.PocketDimensionMap} from {ToPrettyString(uid):pot}");
```
Comment (metalgearsloth) [inline Content.Server/Teleportation/Systems/DimensionPotSystem.cs:39]: Isn't this going to spam warnings on round restart.

File: Content.Shared/Teleportation/Components/PortalBlacklistComponent.cs [hunk 13]
```diff
@@ -0,0 +1,9 @@
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Teleportation.Components;
+
+/// <summary>
+/// Marker component that prevents an entity from using any portals.
+/// </summary>
+[RegisterComponent, NetworkedComponent]
+public sealed class PortalBlacklistComponent : Component { }
```
Comment (metalgearsloth) [inline Content.Shared/Teleportation/Components/PortalBlacklistComponent.cs:9]: Access

File: Content.Shared/Teleportation/Systems/SharedPortalSystem.cs [hunk 14]
```diff
@@ -18,10 +18,11 @@ namespace Content.Shared.Teleportation.Systems;
 /// </summary>
 public abstract class SharedPortalSystem : EntitySystem
 {
-    [Dependency] private readonly IRobustRandom _random = default!;
-    [Dependency] private readonly INetManager _netMan = default!;
     [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly INetManager _netMan = default!;
     [Dependency] private readonly SharedPullingSystem _pulling = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
```
Comment (metalgearsloth) [inline Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:25]: Why re-organise these to a random order?

File: Content.Shared/Teleportation/Systems/SharedPortalSystem.cs [hunk 15]
```diff
@@ -157,6 +142,11 @@ private void OnEndCollide(EntityUid uid, PortalComponent component, ref EndColli
 
         Transform(subject).Coordinates = target;
 
+        // if this portal isn't parented to the grid, do so to prevent funny stuff
+        var grid = target.GetGridUid(EntityManager);
+        if (grid != null)
+            _transform.SetParent(subject, grid.Value);
```
Comment (metalgearsloth) [inline Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:148]: This looks hacky and shouldn't be here.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Catalog/uplink_catalog.yml (hunk 7 above)
Comment (Ilya246) [reply Resources/Prototypes/Catalog/uplink_catalog.yml:302]: why though? the only possible issue with this i see is it being used as overpowered item storage, where you could just use one of the crates at nukie base just as well
alternatively, i could increase the pot's weight so it's slower to drag

File: Resources/Maps/Misc/pocket_dimension.yml (hunk 9 above)
Comment (Ilya246) [reply Resources/Maps/Misc/pocket_dimension.yml:13]: being able to build inside of them is intentional and i, again, don't see how it could cause issues (and if it could, i'd argue the 30TC cost justifies it)

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs (hunk 10 above)
Comment (metalgearsloth) [inline Content.Server/Teleportation/Components/DimensionPotComponent.cs:34]: Also now that I think about it we're gonna have to watch this because if we get more than a couple of these a round need to look at doing something like tabletop because map-spam would be bad.

File: Content.Shared/Teleportation/Systems/SharedPortalSystem.cs (hunk 14 above)
Comment (Ilya246) [reply Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:25]: alphabetic order as opposed to random

Comment (deltanedas) [reply Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:25]: (on field name not type name)

File: Content.Shared/Teleportation/Systems/SharedPortalSystem.cs (hunk 15 above)
Comment (deltanedas) [reply Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:148]: it is absolutely necesaary
if you go out of portal then close it, you get deleted if you dont buckle yourself to something

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs (hunk 10 above)
Comment (deltanedas) [reply Content.Server/Teleportation/Components/DimensionPotComponent.cs:34]: the whole point is to have a unique map, is there much slowdown if a bozo gets 4 of them in a super surplus

Comment (deltanedas) [reply Content.Server/Teleportation/Components/DimensionPotComponent.cs:34]: also how do you delete a map using its entityuid

File: Resources/Prototypes/Catalog/uplink_catalog.yml (hunk 7 above)
Comment (keronshb) [reply Resources/Prototypes/Catalog/uplink_catalog.yml:302]: Nukies do not get this as a tool. It's just cut and dry as that. The blacklist exists for a reason.

File: Resources/Maps/Misc/pocket_dimension.yml (hunk 9 above)
Comment (keronshb) [reply Resources/Maps/Misc/pocket_dimension.yml:13]: It's for balancing reasons. This is required for this to be merged, please add it.

**Comment by AlexMorgan3817** [comment]
Finally, D&D spell implemented in sci-fi
https://www.dndbeyond.com/spells/demiplane

File: Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml (hunk 8 above)
Comment (Ilya246) [reply Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml:276]: would preventing anchoring things inside instead be fine? you would be able to drag things like vendomats inside anyway so i think that would work even better as a solution

File: Content.Shared/Teleportation/Systems/SharedPortalSystem.cs (hunk 14 above)
Comment (AlexMorgan3817) [reply Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:25]: Better sort by type, i guess

File: Resources/Prototypes/Tiles/floors.yml [hunk 16]
```diff
@@ -601,6 +601,20 @@
...
+  baseTurfs:
+  - Plating
+  isSubfloor: true
+  canCrowbar: false
+  footstepSounds:
+    collection: FootstepGrass
+  itemDrop: FloorTileItemGrass
+  isSpace: true
+  heatCapacity: 10000
+
```
Comment (deltanedas) [inline Resources/Prototypes/Tiles/floors.yml:617]: add `indestructible: true` when my pr merge

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml (hunk 8 above)
Comment (keronshb) [reply Resources/Prototypes/Entities/Structures/Furniture/potted_plants.yml:276]: Works for me

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs (hunk 10 above)
Comment (metalgearsloth) [reply Content.Server/Teleportation/Components/DimensionPotComponent.cs]: It makes it a pain in the ass for admins (we already have like 10 maps and it's extremely annoying) and also some stuff is O(n) for map iteration like PVS.

File: Content.Shared/Teleportation/Systems/SharedPortalSystem.cs (hunk 15 above)
Comment (metalgearsloth) [reply Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:148]: If you have the same parent as the portal then deleting the portal shouldn't delete you unless there's some other bug going on, in which case this would still be a bandaid.

File: Content.Server/Teleportation/Components/DimensionPotComponent.cs (hunk 10 above)
Comment (deltanedas) [reply Content.Server/Teleportation/Components/DimensionPotComponent.cs]: would pvs be better if it was a single map being loadgrid'd with the pot interior thing

Comment (metalgearsloth) [reply Content.Server/Teleportation/Components/DimensionPotComponent.cs]: No?

File: Content.Shared/Teleportation/Systems/SharedPortalSystem.cs (hunk 15 above)
Comment (deltanedas) [reply Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:148]: shouldnt deleting any entity by design delete its children?

Comment (metalgearsloth) [reply Content.Shared/Teleportation/Systems/SharedPortalSystem.cs:148]: Yeah?

So what happens atm:
Subject coordinates set to target coordinates (expensive).
If target on a grid now set the subject's coordinates to grid (expensive).

If you just want to attach it to the grid then do that directly.

**Comment by metalgearsloth** [comment]
Re-open if you come back to this.
